recipes/libffi/all/conanfile.py

```diff
@@ -22,8 +22,26 @@ class LibffiConan(ConanFile):
         "fPIC": True,
     }
     exports_sources = "patches/**"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
     _autotools = None
-    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
```

Comment:
If methods are reordered, what not moving build_requirements() after configure()?

Reply:
do you think the order of methods matters that much? IMO it's just personal preference if you want to have alphanumeric order of methods or something else.

Reply:
understanding order of execution is important to avoid issues like options not available in subsequent methods or already deleted in package_id, understand that what happen in some methods don't contribute to package_id, or should be taken into account in package_info. Ordering methods by order of execution helps to understand the flow of execution of a recipe.

This PR modifies order of methods, and it clearly reorders methods by order of execution (config_options() and configure() moved before source()), except build_requirements(), I was just higlighting this point.

Reply:
I indeed tried to reorder the methods as said by @SpaceIm .
Let's move `build_requirements` when the next package is missing :smile:

Reply:
Please don't jinx us...

Abandon ship! 🌊 ⛵